ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py

```diff
@@ -206,7 +206,19 @@ def get_column_tag_labels(
         return None
 
     def set_inspector(self, database_name: str):
-        self.client = Client(project=database_name)
+        # TODO support location property in JSON Schema
+        # TODO support OAuth 2.0 scopes
+        impersonate_service_account = (
+            self.service_connection.credentials.gcpImpersonateServiceAccount.impersonateServiceAccount
+        )
+        lifetime = (
+            self.service_connection.credentials.gcpImpersonateServiceAccount.lifetime
+        )
```

Comment:
can we add a check here to check if the connectiontype is selected as `gcpImpersonateServiceAccount` and then get the attributes from that instead. or else it would be throw an error and ingestion would fail if the user selects another type